src/modules/tools/zprobe/ZGridStrategy.cpp

```diff
@@ -251,20 +247,30 @@ bool ZGridStrategy::handleGcode(Gcode *gcode)
 
             // M374: Save grid
             case 374:{
-                if(this->saveGrid()) {
+               char gridname[5];
+
+               if(gcode->has_letter('S'))  // Custom grid number
+                    sprintf(gridname, "S%3.0f", gcode->get_value('S'));
```

Comment:
This is too dangerous you will get stack corruption and crash if someone enter M374 S123456
You must use snprintf(gridname, 4, ...)